README.md

```diff
@@ -1,2 +1,88 @@
-# icepy
-Python-esque programming language with Icelandic grammar rules
+# IcePy
+
+A programming language that has strict Icelandic grammar rules built in.
+
+## Concepts
+
+There are a set of known variable names that can be used. Each have forms in
+nominative, accusative, dative, genitive.
+
+Each argument to a function must define the case expected.
+
+Each class has a gender
+
+Core constructs of the language may take different cases too.
+
+
+## Language constructs
+
+As the name suggests, the language is python-esque. Except that there are mass
+translations of things (eg `class` and `def` are no longer available).
+
+
+### Defining a method or function
+
+To define the function "reikna", you use the keyword "virka"
+
+```
+virka reikna():
+    pass
+```
+
+### Defining a class (tegund)
+
+```
+tegund Bátur():
+
+   einstaklingar = []
+
+   virka __frumstilla__(sjalfur):
+       pass
+
+```
+
+### Iterating through an iterable
+
+```
+hópur = ['Margrét', 'Anna', 'Emma', 'Þór', 'Hrafn', 'Freyr']
+
+fyrir einstaklingur í hóp:
+    # Gera eitthvað
+
+```
+Note:
+ - that using `fyrir einstaklingur í hópur` would raise a syntax error here, on
+   account of hópur not being in the correct case.
+ - that the days of `for i in x` are behind you now. Even these somewhat
+   temporary variable names must be known to the system - so that it can
+   determine case when they are used.
+
+
+### Calling methods with correct case
+
+Note that all variables should be instansiated in nominative form. Once
+created they are available to the interpreter in any of the cases and must be
+used appropriately.
+
+```
+bátur = Bátur()
+bátur.
```

Comment:
I'm looking here for a method that could be called on a boat that would require an object to be given in some non-nominative form.

Icepy will expose developers to the same mental hoop-jumping that one must go through in order to speak Icelandic.

In this case, I'm after something like this, but with translations: 
```python
einstaklingur = 'alex'
bátur.some_method_that_takes_accusative(einstakling)
```

This illustrates that upon defining a variable, the interpreter has all forms of that variable available to it.

The method definition would look something like

```
virka some_method_that_takes_accusative(nafn: str:þf)
```

and the interpreter raises if a non þf-form variable name is passed.

> not really sure how practical this is.

You're right, this will be completely impractical.